### PR TITLE
PSR-1324 - Declaration redirect

### PR DESCRIPTION
**Pull request by kberec**

### CODE REVIEW

**Review by ccatosdevelopment — DISMISSED** [review]
The banner is displaying and data is being wiped in an instance where it shouldn't, so I think this logic needs reworking.

If the previously submitted return is a full return, and >99 members are added in this return, and all sections are completed, when you get to the declaration page, the notification banner will be displayed and the data will be wiped, even though this is a full return and not a skipped one.

File: test/utils/nonsipp/MemberCountUtilsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import utils.UserAnswersUtils.UserAnswersOps
+import org.scalatest.OptionValues
+import generators.ModelGenerators.pensionSchemeIdGen
+import models._
+import pages.nonsipp.moneyborrowed._
+import pages.nonsipp.bonds._
+import pages.nonsipp.memberdetails._
+import org.scalatest.freespec.AnyFreeSpec
+import pages.nonsipp.loansmadeoroutstanding._
+import models.IdentitySubject._
```
Comment (cyeniceri) [inline test/utils/nonsipp/MemberCountUtilsSpec.scala:36]: it seems this import is unused

File: app/controllers/nonsipp/declaration/PsaDeclarationController.scala [hunk 2]
```diff
@@ -73,11 +93,20 @@ class PsaDeclarationController @Inject()(
           val now = schemeDateService.now()
 
           for {
-            _ <- psrSubmissionService.submitPsrDetails(
-              srn = srn,
-              isSubmitted = true,
-              fallbackCall = controllers.nonsipp.declaration.routes.PsaDeclarationController.onPageLoad(srn)
-            )
+            journeyByPassed <- isJourneyBypassed(srn)
+            bypassed = journeyByPassed.getOrElse(false)
```
[the remark this answers is not in the document]
Comment (kberec) [reply app/controllers/nonsipp/declaration/PsaDeclarationController.scala:97]: Yes, I think it is safe to ignore the journeyRecovery here. I'm just re-using some code here and the recovery outcome is more relevant in the places where this was originally used.

File: app/controllers/nonsipp/declaration/PspDeclarationController.scala [hunk 3]
```diff
@@ -99,11 +117,20 @@ class PspDeclarationController @Inject()(
                   updatedAnswers <- Future
                     .fromTry(request.userAnswers.set(PspDeclarationPage(srn), answer))
                   _ <- saveService.save(updatedAnswers)
-                  _ <- psrSubmissionService.submitPsrDetails(
-                    srn = srn,
-                    isSubmitted = true,
-                    fallbackCall = controllers.nonsipp.declaration.routes.PspDeclarationController.onPageLoad(srn)
-                  )
+                  journeyByPassed <- isJourneyBypassed(srn)
+                  bypassed = journeyByPassed.getOrElse(false)
```
[the remark this answers is not in the document]
Comment (kberec) [reply app/controllers/nonsipp/declaration/PspDeclarationController.scala:121]: Same answer as above :)

File: test/utils/nonsipp/MemberCountUtilsSpec.scala (hunk 1 above)
Comment (kberec) [reply test/utils/nonsipp/MemberCountUtilsSpec.scala:36]: Can I remove this in my next branch, to spare the pipeline time?

Comment (cyeniceri) [reply test/utils/nonsipp/MemberCountUtilsSpec.scala:36]: I can also do

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**